example/src/test/java/com/ryanharter/auto/value/gson/example/PersonTest.java

```diff
@@ -0,0 +1,30 @@
+package com.ryanharter.auto.value.gson.example;
+
+import com.google.gson.Gson;
+import com.google.gson.GsonBuilder;
+import org.junit.Assert;
+import org.junit.Test;
+import org.skyscreamer.jsonassert.JSONAssert;
+
+public class PersonTest {
+    @Test
+    public void testGson() throws Exception {
+        Gson gson = new GsonBuilder()
+                .registerTypeAdapterFactory(new PersonTypeAdapterFactory())
+                .create();
+        Person person = Person.builder()
+                .name("Piasy")
+                .gender(1)
+                .age(23)
+                .build();
+        String json = "{\"name\":\"Piasy\",\"age\":23,\"gender\":1}";
+
+        String toJson = gson.toJson(person, Person.class);
+        JSONAssert.assertEquals(json, toJson, false);
```

Comment:
It should be deterministic based on declaration order of fields.

On Tue, Apr 19, 2016, 8:25 PM Piasy <notifications@github.com> wrote:

> In
> example/src/test/java/com/ryanharter/auto/value/gson/example/PersonTest.java
> <https://github.com/rharter/auto-value-gson/pull/30#discussion_r60334804>:
>
> > +
> > +public class PersonTest {
> > +    @Test
> > +    public void testGson() throws Exception {
> > +        Gson gson = new GsonBuilder()
> > +                .registerTypeAdapterFactory(new PersonTypeAdapterFactory())
> > +                .create();
> > +        Person person = Person.builder()
> > +                .name("Piasy")
> > +                .gender(1)
> > +                .age(23)
> > +                .build();
> > +        String json = "{\"name\":\"Piasy\",\"age\":23,\"gender\":1}";
> > +
> > +        String toJson = gson.toJson(person, Person.class);
> > +        JSONAssert.assertEquals(json, toJson, false);
>
> would toJson return string with different field order with our provided
> string?
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/rharter/auto-value-gson/pull/30/files/e2d8ec9e7411524859dd3b2cfe98a76e6f3ce9e5#r60334804>
>